utils/eeprom_metadata.py

```diff
@@ -131,15 +147,34 @@ def getArguments():
     application_name = args['appname']
     output_file = args['output']
     print_header_result = args['verbose']
-    return binary_file, application_name, output_file, print_header_result
-
+    dont_auto_convert = args['dont_convert']
+    return binary_file, application_name, output_file, print_header_result, dont_auto_convert
+
+def check_if_binary(check):
+    file_name, file_ext = splitext(check)
+    if file_ext == ".bin":
+        return True, check
+    elif file_ext == ".hex":
+        if not dont_auto_convert:
+            # check=True is used to raise an exception if the command does not run successfully (doesn't exist, bad file, etc)
```

Comment:
you don't need this comment here. If you still think you need to tell the user your intent, please refactor this to a new free function.

Reply:
Apologies, I tend to over-comment :)